lib/cloud-metadata/gcp.js

```diff
@@ -44,11 +48,25 @@ function getMetadataGcp(
     });
 
     res.on('end', function (data) {
+      if (res.statusCode !== 200) {
+        logger.debug('gcp metadata: unexpected statusCode: %s', res.statusCode);
+        cb(
+          new Error(
+            'error fetching gcp metadata: unexpected statusCode: ' +
+              res.statusCode,
+          ),
+        );
+        return;
+      }
+      // Note: We could also guard on the response having the
+      // 'Metadata-Flavor: Google' header as done by:
+      // https://github.com/googleapis/gcp-metadata/blob/v6.0.0/src/index.ts#L109-L112
+
```

Comment:
Reviewer note: Add a sanity guard on the response from the Google metadata API. If it isn't a 200, we shouldn't get into parsing it.  This is my justification for dropping some of the (unnecessary, IMO) defensive coding and testing cases.

Reply:
> This is my justification for dropping some of the (unnecessary, IMO) defensive coding and testing cases.

I agree that we can assume safely that a 200 status comes with the proper metadata described in their docs. Also it simplifies the code so IMO this is a good change :)